### ISSUE TITLE
Image component doesn't update with hooks

### ISSUE THREAD

**Issue by salvini29** [GeekyAnts/NativeBase#4518]
The Image component behaves differently from React Native's Image component. If I update a hook that changes the source of an Image of NativeBase the component doesn't update but in React Native the component updates and the Image changes.

**Code for React Native Image component(the Image changes):**

```
import React, { useState, useEffect } from 'react';
import { View, Image, Button} from 'react-native';

const img1 = require('../src/img/img1.png');
const img2 = require('../src/img/img2.png');
const images = {img1,img2};

const Example = () => {

  const [selected, setSelected] = useState(images.img1);

  return (
    <View>
      <Image
        source={selected}
      />
      <Button title="Change Image" onPress={() => {setSelected(images.img2)}}/>
    </View>
  );
}

export default Example;
```

**Code for NativeBase Image component(the Image doesn't changes):**

```
import React, { useState, useEffect } from 'react';
import { Image, Button, NativeBaseProvider } from "native-base";

const img1 = require('../src/img/img1.png');
const img2 = require('../src/img/img2.png');
const images = {img1,img2};

const Example = () => {

   const [selected, setSelected] = useState(images.img1);

  return (
    <NativeBaseProvider>
      <Image
        source={selected}
        alt="image"
      />
      <Button onPress={() => {setSelected(images.img2)}}>Change Image</Button>
    </NativeBaseProvider>
  );
}

export default Example;
```

I'm using: 
native-base: ^3.2.2,
react: 17.0.2,
react-native: 0.65.0

On a Pixel 3A emulated phone.

**Comment by mayank-96** [comment]
Hi @salvini29, thanks for reporting this issue. We'll look into this.

**Comment by mayank-96** [comment]
Hello @salvini29, this issue has been fixed in v3.3.2.
